File: 102. Binary Tree Level Order Traversal/sol.go
package levelOrderTraversal

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func levelOrder(root *TreeNode) [][]int {
	res := make([][]int, 0)
	if root == nil {
		return res
	}
	queue := []*TreeNode{root}
	for len(queue) > 0 {
		queLen := len(queue)
		level := make([]int, 0)
		for i := range queLen {
			level = append(level, queue[i].Val)
			if queue[i].Left != nil {
				queue = append(queue, queue[i].Left)
			}
			if queue[i].Right != nil {
				queue = append(queue, queue[i].Right)
			}
		}
		queue = queue[queLen:]
		res = append(res, level)
	}

	return res

}

File: 102. Binary Tree Level Order Traversal/sol.py
from typing import Optional 
class TreeNode:
    def __init__(self, val: int = 0, left: 'Optional[TreeNode]' = None, right: 'Optional[TreeeNode]' = None):
        self.val = val 
        self.left = val
        self.right = right 
        
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        if not root:
            return res

        stack = [[root]]
        while len(stack) > 0 :
            nodes = stack.pop()
            currValues = []
            currStack = []
            for node in nodes:
                currValues.append(node.val)
                if node.left:
                    currStack.append(node.left)
                if node.right:
                    currStack.append(node.right)
            if len(currValues) == 0:
                break
            res.append(currValues)
            stack.append(currStack)
        return res

        